spring-cloud-task-docs/src/main/asciidoc/getting-started.adoc

```diff
@@ -0,0 +1,276 @@
+=== Adding classpath dependencies
+
+A Spring Cloud Task is made up of a Spring Boot application that is expected to end.  In
+our POM above, we created the shell of a Spring Boot application from a dependency
+perspective by setting our parent to use the `spring-boot-starter-parent`.
+
+Spring Boot provides a number of additional "Starter POMs".  Some of which are appropriate
+for use within tasks (`spring-boot-starter-batch`, `spring-boot-starter-jdbc`, etc) and
+some may not be ('spring-boot-starter-web` is probably not going to be used in a task).
+The indicator of if a starter makes sense or not comes down to if the resulting
+application will end (batch based applications typically end, the
+`spring-boot-starter-web` dependency bootstraps a servlet container which probably wont').
+
+For this example, we'll only need to add a single additional dependency, the one for
+Spring Cloud Task itself:
+
+[source,xml]
+		<dependency>
+			<groupId>org.springframework.cloud</groupId>
+			<artifactId>spring-cloud-task-core</artifactId>
+			<version>1.0.0.BUILD-SNAPSHOT</version>
+		</dependency>
+
+[[getting-started-writing-the-code]]
+=== Writing the code
+
+To finish our application, we need to create a single Java file.  Maven will compile the
+sources from `src/main/java` by default so you need to create that folder structure.  Then
+add a file named `src/main/java/com/example/SampleTask.java`:
+
+[source,java]
+----
+package com.example;
+
+import org.springframework.boot.*;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.task.configuration.EnableTask;
+import org.springframework.context.annotation.Bean;
+
+@SpringBootApplication
+@EnableTask
+public class SampleTask {
+
+	@Bean
+	public CommandLineRunner commandLineRunner() {
+		return new HelloWorldCommandLineRunner();
+	}
+
+	public static void main(String[] args) {
+		SpringApplication.run(SampleTask.class, args);
+	}
+
+	public static class HelloWorldCommandLineRunner implements CommandLineRunner {
+
+		@Override
+		public void run(String... strings) throws Exception {
+			System.out.println("Hello World!");
+		}
+	}
+}
+----
+
+While it may not look like much, quite a bit is going on.  To read more about the Spring
+Boot specifics, take a look at their reference documentation here:
+http://docs.spring.io/spring-boot/docs/current/reference/html/[http://docs.spring.io/spring-boot/docs/current/reference/html/]
+
+[[getting-started-at-task]]
+==== The @EnableTask annotation
+
+The first non boot annotation in our example is the `@EnableTask` annotation.  This class
+level annotation tells Spring Cloud Task to bootstrap it's functionality.  This occurs by
+importing an additional configuration class, `SimpleTaskConfiguration` by default.  This
+additional configuration registers the `TaskRepository` and the infrastructure for its
+use.
+
+Out of the box, the `TaskRepository` will use an in memory database to record the results
```

Comment:
If the user doesn't specify the 	`spring-boot-starter-jdbc` the Map will be used.